### ISSUE TITLE
Task is still running even used publisher plugin has been unloaded

### ISSUE THREAD

**Issue by IzabellaRaulin** [intelsdi-x/snap#1659]
At present, there is a possibility to unload the plugin even if it is used by running task.  I used [exemplary task manifest](https://github.com/intelsdi-x/snap-plugin-collector-cpu/blob/master/examples/tasks/cpu-file.json) (collecting cpu metrics and publishing them to a file) to verify is that behavior is what is really expected.

*Scenario A: unloading collector plugin:*
1) Load cpu and file plugins. Create and start the task 
2) Unload collector plugin: `snaptel plugin unload collector cpu 6`
Observation:
Receiving errors i.a. " Metric not found: /intel/procfs/cpu/*/guest_jiff:". After 10 Fails, task status changes from `Running` to `Disabled`

*Scenario B: with unloading publisher plugin:*
1) Load cpu and file plugins. Create and start the task 
2) Unload publisher plugin: `snaptel plugin unload publisher file 2`
Observation:
No errors, no misses. The task has status _Running_. Metrics are still publishing to file.
3) Stop the task and try to re-start it
    ```
    snaptel task start 94969d7a-cd0c-4f6d-8281-109e3977b0b6
    Error starting task:
    error 0: Plugin not found: type(publisher) name(file) version(-1)
    ```

Basically, **the behavior should be similar in both cases** - this is what I expect. 
The another option which might be taken into consideration is **not allowing for unloading the plugin which is used in a running task**.
